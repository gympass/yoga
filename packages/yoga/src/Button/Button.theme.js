import { hexToRgb } from '@gympass/yoga-common';

const Button = ({
  spacing,
  fontSizes,
  borders,
  radii,
  colors,
  fontWeights,
}) => ({
  padding: {
    default: {
      right: spacing.large,
      left: spacing.large,
    },
    small: {
      right: spacing.small,
      left: spacing.small,
    },
  },
  height: {
    default: 48,
    small: 32,
  },
  font: {
    size: {
      default: fontSizes.medium,
      small: fontSizes.small,
    },
    weight: fontWeights.medium,
  },
  border: {
    radius: radii.circle,
  },
  icon: {
    size: {
      default: 24,
      small: 16,
    },
    margin: {
      right: spacing.xxsmall,
    },
  },
  types: {
    contained: {
      backgroundColor: {
        disabled: colors.elements.backgroundAndDisabled,
        primary: {
          default: colors.primary,
          pressed: hexToRgb(colors.primary, 0.75),
        },
        secondary: {
          default: colors.secondary,
          pressed: hexToRgb(colors.secondary, 0.75),
        },
      },
      font: {
        default: {
          color: colors.white,
        },
        disabled: {
          color: colors.text.disabled,
        },
        pressed: {
          color: colors.white,
        },
      },
    },
    outline: {
      backgroundColor: {
        default: 'transparent',
        primary: {
          hover: colors.primary,
        },
        secondary: {
          hover: colors.secondary,
        },
      },
      border: {
        width: borders.small,
      },
      font: {
        default: {
          primary: {
            color: colors.primary,
          },
          secondary: {
            color: colors.secondary,
          },
        },
        disabled: {
          color: colors.deep,
        },
        hover: {
          color: colors.white,
        },
        pressed: {
          primary: {
            color: hexToRgb(colors.primary, 0.75),
          },
          secondary: {
            color: hexToRgb(colors.secondary, 0.75),
          },
        },
      },
    },
    text: {
      backgroundColor: 'transparent',
      disabled: colors.deep,
    },
    link: {
      font: {
        primary: {
          color: colors.primary,
        },
        secondary: {
          color: colors.secondary,
        },
        disabled: {
          color: colors.deep,
        },
      },
      margin: {
        top: spacing.small,
        bottom: spacing.xxsmall,
      },
      icon: {
        margin: {
          right: spacing.xxxsmall,
        },
      },
    },
  },
});

export default Button;
