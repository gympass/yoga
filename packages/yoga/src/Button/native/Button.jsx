import React, { forwardRef } from 'react';
import { bool, func, node, oneOfType } from 'prop-types';
import styled, { withTheme } from 'styled-components';

import withTouchable from './withTouchable';

import Text from '../../Text';

export const Label = styled(Text).attrs(
  ({
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => ({
    fontWeight: button.font.weight,
  }),
)`
  text-align: center;
  ${({
    small,
    color,
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => `
      font-size: ${small ? button.font.size.small : button.font.size.default}px;
      line-height: ${
        small ? button.font.lineHeight.small : button.font.lineHeight.default
      }px;
      color: ${color};
    `}
`;

export const ButtonContainer = styled.View`
  ${({
    pressed,
    disabled,
    full,
    small,
    inverted,
    secondary,
    theme: {
      yoga: {
        components: { button },
      },
    },
  }) => {
    const state = secondary ? 'secondary' : 'primary';

    return `
      flex-direction: row;
      align-items: center;
      justify-content: center;

      background-color: ${
        button.types.contained.backgroundColor[state].default
      };
      border-radius: ${button.border.radius}px;
      height: ${small ? button.height.small : button.height.default};
      justify-content: center;
      padding-left: ${
        small ? button.padding.small.left : button.padding.default.left
      }px;
      padding-right: ${
        small ? button.padding.small.right : button.padding.default.right
      }px;
      ${full ? 'width: 100%;' : ''}

      ${
        !disabled && pressed
          ? `
        background-color: ${button.types.contained.backgroundColor[state].pressed};
      `
          : ''
      }

      ${
        disabled
          ? `
        background-color: ${button.types.contained.backgroundColor.disabled};
      `
          : ''
      }

      ${
        inverted && !disabled
          ? `
          background-color: ${button.types.contained.font.default.color};
        `
          : ''
      }
    `;
  }}
`;

/** Buttons make common actions more obvious and help users more easily perform
them. Buttons use labels and sometimes icons to communicate the action that will
occur when the user touches them. */
const Button = forwardRef(
  (
    {
      children = 'Button',
      full = false,
      disabled = false,
      small = false,
      pressed = false,
      inverted = false,
      secondary = false,
      icon: Icon,
      theme: {
        yoga: {
          components: { button },
        },
      },
      ...rest
    },
    ref,
  ) => {
    const state = secondary ? 'secondary' : 'primary';
    let textColor = button.types.contained.font.default.color;

    if (disabled) {
      textColor = button.types.contained.font.disabled.color;
    } else if (inverted) {
      textColor = button.types.contained.backgroundColor[state].default;
      if (pressed) {
        textColor = button.types.contained.backgroundColor[state].pressed;
      }
    } else if (pressed) {
      textColor = button.types.contained.font.pressed.color;
    }

    return (
      <ButtonContainer
        {...rest}
        full={full}
        pressed={pressed}
        disabled={disabled}
        small={small}
        inverted={inverted}
        secondary={secondary}
        ref={ref}
      >
        {Icon && (
          <Icon
            width={small ? button.icon.size.small : button.icon.size.default}
            height={small ? button.icon.size.small : button.icon.size.default}
            fill={textColor}
            style={{
              marginRight: button.icon.margin.right,
            }}
          />
        )}
        <Label
          disabled={disabled}
          pressed={pressed}
          inverted={inverted}
          small={small}
          color={textColor}
        >
          {children}
        </Label>
      </ButtonContainer>
    );
  },
);

Button.propTypes = {
  children: node,
  full: bool,
  disabled: bool,
  small: bool,
  pressed: bool,
  inverted: bool,
  secondary: bool,
  /** an Icon from yoga-icons package */
  icon: oneOfType([bool, func]),
};

const ButtonWithTouchable = withTouchable(withTheme(Button));

export default ButtonWithTouchable;
