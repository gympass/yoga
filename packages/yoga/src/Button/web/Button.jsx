import React, { forwardRef } from 'react';
import { func, node, oneOfType, bool, string } from 'prop-types';
import styled from 'styled-components';
import StyledButton from './StyledButton';
import Spinner from '../../Spinner';

const SpinnerContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

const noop = () => {};

/** Buttons make common actions more obvious and help users more easily perform them. Buttons use labels and sometimes icons to communicate the action that will occur when the user touches them. */
const Button = forwardRef(
  (
    {
      children = 'Button',
      onClick = noop,
      full = false,
      disabled = undefined,
      inverted = false,
      small = false,
      secondary = false,
      icon: Icon,
      isLoading = false,
      'aria-label': ariaLabel,
      ...props
    },
    ref,
  ) => {
    const finalProps = {
      ...props,
    };

    if (props.href) {
      finalProps.as = 'a';
    }

    return (
      <StyledButton
        ref={ref}
        disabled={disabled || isLoading}
        aria-disabled={disabled}
        full={full}
        inverted={inverted}
        onClick={onClick}
        small={small}
        secondary={secondary}
        isLoading={isLoading}
        aria-label={ariaLabel}
        {...finalProps}
      >
        {Icon && <Icon role="img" />}
        {children}

        {isLoading && (
          <SpinnerContainer>
            <Spinner color="text.disabled" size={small ? 'small' : 'medium'} />
          </SpinnerContainer>
        )}
      </StyledButton>
    );
  },
);

Button.propTypes = {
  ariaLabel: string,
  children: node,
  disabled: bool,
  full: bool,
  inverted: bool,
  isLoading: bool,
  onClick: func,
  small: bool,
  secondary: bool,
  /** an Icon from yoga-icons package */
  icon: oneOfType([node, func]),
  href: string,
  'aria-label': string,
};

Button.displayName = 'Button';

export default Button;
