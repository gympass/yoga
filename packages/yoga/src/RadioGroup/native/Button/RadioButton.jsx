import React, { useContext } from 'react';
import { string, oneOfType, number, node } from 'prop-types';
import { TouchableWithoutFeedback } from 'react-native';
import styled from 'styled-components';

import RadioGroupContext from '../../RadioGroupContext';

const RadioMark = styled.View(
  ({
    theme: {
      yoga: {
        components: { radiogroup },
      },
    },
    small,
    checked,
  }) => `
  align-items: center;
  justify-content: center;

  height: ${
    small ? radiogroup.button.height.small : radiogroup.button.height.normal
  }px;
  padding-left: ${radiogroup.button.padding.left}px;
  padding-right: ${radiogroup.button.padding.right}px;

  background-color: ${radiogroup.button.backgroundColor};
  border-radius: ${radiogroup.button.border.radius}px;

  font-size: ${radiogroup.font.size}px;

  ${
    checked
      ? `
          background-color: ${radiogroup.checked.backgroundColor};
        `
      : ``
  }
`,
);

const Text = styled.Text(
  ({
    theme: {
      yoga: {
        baseFont,
        components: { radiogroup },
      },
    },
    checked,
  }) => `
    font-family: ${baseFont.family};
    color: ${radiogroup.font.color};

    ${
      checked
        ? `
          color: ${radiogroup.checked.font.color};
          font-weight: ${radiogroup.checked.font.weight};
        `
        : ''
    }
    `,
);

/** Radio group allows user to select one option from a set of options. Use
value on group to set the selected option. */
const RadioGroupButton = React.forwardRef(
  ({ value = '', children, ...rest }, ref) => {
    const { onChange, small, selectedValue, ...context } =
      useContext(RadioGroupContext);

    const inputValue = value || children;
    const checked = inputValue === selectedValue;

    return (
      <TouchableWithoutFeedback
        onPress={() => {
          onChange({ value: inputValue });
        }}
        {...context}
      >
        <RadioMark small={small} checked={checked} ref={ref} {...rest}>
          <Text checked={checked}>{children}</Text>
        </RadioMark>
      </TouchableWithoutFeedback>
    );
  },
);

RadioGroupButton.displayName = 'RadioGroup.Button';

RadioGroupButton.propTypes = {
  value: oneOfType([string, number]),
  children: node.isRequired,
};

export default RadioGroupButton;
