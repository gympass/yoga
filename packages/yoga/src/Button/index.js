import Button from './web/Button';
import ButtonOutline from './web/Outline';
import ButtonText from './web/Text';
import ButtonLink from './web/Link';
import ButtonIcon from './web/Icon';

const ExportButton = Object.assign(Button, {
  Outline: ButtonOutline,
  Text: ButtonText,
  Link: ButtonLink,
  Icon: ButtonIcon,
});

export default ExportButton;
