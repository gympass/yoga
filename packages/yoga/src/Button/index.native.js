import Button from './native/Button';
import ButtonText from './native/Text';
import ButtonLink from './native/Link';
import ButtonIcon from './native/Icon';

const ExportButton = Object.assign(Button, {
  Text: ButtonText,
  Link: ButtonLink,
  Icon: ButtonIcon,
});

export default ExportButton;
