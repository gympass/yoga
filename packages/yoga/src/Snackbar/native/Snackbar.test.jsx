import React, { useRef } from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import * as RN from 'react-native';

import { CheckedFull } from '@gympass/yoga-icons';
import { ThemeProvider, Snackbar } from '../..';
import Button from '../../Button';

RN.Animated.spring = jest.fn().mockReturnValue({
  start: jest.fn(),
});

const Component = overrideProps => {
  const snackbarRef = useRef();

  return (
    <ThemeProvider>
      <Button
        full
        small
        onPress={() => {
          snackbarRef.current.open();
        }}
      >
        Tap to open snackbar
      </Button>
      <Snackbar
        ref={snackbarRef}
        message="Feedback Message"
        {...overrideProps}
      />
    </ThemeProvider>
  );
};

describe('<Snackbar />', () => {
  it('should match snapshot when snackbar is default', () => {
    const { toJSON } = render(<Component />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when have an icon and action', () => {
    const overrideProps = {
      icon: CheckedFull,
      actionLabel: 'Action',
      onAction: () => {},
    };

    const { toJSON } = render(<Component {...overrideProps} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when have a long text', () => {
    const overrideProps = {
      message: 'Lorem Ipsum is simply dummy text of the printing and types.',
      icon: CheckedFull,
      actionLabel: 'Action',
      onAction: () => {},
    };

    const { toJSON } = render(<Component {...overrideProps} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when have a variant informative or attention prop', () => {
    const overrideProps = {
      variant: 'informative',
    };

    const { toJSON } = render(<Component {...overrideProps} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show and hide snackbar when the open and close Animation function is called', () => {
    const overrideProps = {
      onSnackbarClose: jest.fn(),
      actionLabel: 'Action',
      onAction: jest.fn(),
    };

    const { getByText, getByTestId } = render(<Component {...overrideProps} />);

    fireEvent(getByTestId('wrapper'), 'layout', {
      nativeEvent: {
        layout: { height: 200 },
      },
    });

    fireEvent.press(getByText('Tap to open snackbar'));

    expect(RN.Animated.spring.mock.calls[0][1]).toEqual({
      toValue: 0,
      bounciness: 0,
      useNativeDriver: true,
    });

    fireEvent.press(getByText('Action'));

    expect(RN.Animated.spring.mock.calls[1][1]).toEqual({
      toValue: 200,
      bounciness: 0,
      useNativeDriver: true,
    });
  });
});
