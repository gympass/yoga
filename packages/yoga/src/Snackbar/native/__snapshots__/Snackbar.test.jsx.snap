// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should match snapshot when have a long text 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      secondary={false}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#D8385E",
            "borderRadius": 9999,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 16,
            "paddingRight": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        color="#FFFFFF"
        disabled={false}
        inverted={false}
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "500",
              "textAlign": "center",
            },
          ]
        }
      >
        Tap to open snackbar
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <View
      onLayout={[Function]}
      testID="wrapper"
    >
      <View
        bottomOffset={0}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#C1EEDB",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 20,
              "marginHorizontal": 20,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 12,
            },
          ]
        }
        variant="success"
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          fill="#231B22"
          focusable={false}
          height={24}
          marginRight="xxsmall"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {
                  "height": 24,
                  "marginRight": 8,
                  "width": 24,
                },
              ],
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          width={24}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4280490786,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
        <Text
          flex={1}
          fontSize="small"
          inverted={false}
          light={false}
          marginVertical="xxxsmall"
          numberOfLines={2}
          size="medium"
          style={
            Array [
              Object {
                "color": "#231B22",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
                "marginBottom": 4,
                "marginTop": 4,
              },
            ]
          }
        >
          Lorem Ipsum is simply dummy text of the printing and types.
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            marginLeft="xxsmall"
            pressed={false}
            small={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 9999,
                  "flexDirection": "row",
                  "height": 32,
                  "justifyContent": "center",
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "marginLeft": 8,
                },
              ]
            }
          >
            <Text
              color="#231B22"
              pressed={false}
              secondary={true}
              small={true}
              style={
                Array [
                  Object {
                    "color": "#231B22",
                    "fontFamily": "Rubik",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Action
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`<Snackbar /> should match snapshot when have a variant informative or attention prop 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      secondary={false}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#D8385E",
            "borderRadius": 9999,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 16,
            "paddingRight": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        color="#FFFFFF"
        disabled={false}
        inverted={false}
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "500",
              "textAlign": "center",
            },
          ]
        }
      >
        Tap to open snackbar
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <View
      onLayout={[Function]}
      testID="wrapper"
    >
      <View
        bottomOffset={0}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E0DFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 20,
              "marginHorizontal": 20,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 12,
            },
          ]
        }
        variant="informative"
      >
        <Text
          flex={1}
          fontSize="small"
          inverted={false}
          light={false}
          marginVertical="xxxsmall"
          numberOfLines={2}
          size="medium"
          style={
            Array [
              Object {
                "color": "#231B22",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
                "marginBottom": 4,
                "marginTop": 4,
              },
            ]
          }
        >
          Feedback Message
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`<Snackbar /> should match snapshot when have an icon and action 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      secondary={false}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#D8385E",
            "borderRadius": 9999,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 16,
            "paddingRight": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        color="#FFFFFF"
        disabled={false}
        inverted={false}
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "500",
              "textAlign": "center",
            },
          ]
        }
      >
        Tap to open snackbar
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <View
      onLayout={[Function]}
      testID="wrapper"
    >
      <View
        bottomOffset={0}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#C1EEDB",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 20,
              "marginHorizontal": 20,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 12,
            },
          ]
        }
        variant="success"
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          fill="#231B22"
          focusable={false}
          height={24}
          marginRight="xxsmall"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {
                  "height": 24,
                  "marginRight": 8,
                  "width": 24,
                },
              ],
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          width={24}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4280490786,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGSvgView>
        <Text
          flex={1}
          fontSize="small"
          inverted={false}
          light={false}
          marginVertical="xxxsmall"
          numberOfLines={2}
          size="medium"
          style={
            Array [
              Object {
                "color": "#231B22",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
                "marginBottom": 4,
                "marginTop": 4,
              },
            ]
          }
        >
          Feedback Message
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            marginLeft="xxsmall"
            pressed={false}
            small={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 9999,
                  "flexDirection": "row",
                  "height": 32,
                  "justifyContent": "center",
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "marginLeft": 8,
                },
              ]
            }
          >
            <Text
              color="#231B22"
              pressed={false}
              secondary={true}
              small={true}
              style={
                Array [
                  Object {
                    "color": "#231B22",
                    "fontFamily": "Rubik",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Action
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`<Snackbar /> should match snapshot when snackbar is default 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      disabled={false}
      full={true}
      inverted={false}
      pressed={false}
      secondary={false}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#D8385E",
            "borderRadius": 9999,
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 16,
            "paddingRight": 16,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        color="#FFFFFF"
        disabled={false}
        inverted={false}
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "500",
              "textAlign": "center",
            },
          ]
        }
      >
        Tap to open snackbar
      </Text>
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <View
      onLayout={[Function]}
      testID="wrapper"
    >
      <View
        bottomOffset={0}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#C1EEDB",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 20,
              "marginHorizontal": 20,
              "paddingBottom": 12,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 12,
            },
          ]
        }
        variant="success"
      >
        <Text
          flex={1}
          fontSize="small"
          inverted={false}
          light={false}
          marginVertical="xxxsmall"
          numberOfLines={2}
          size="medium"
          style={
            Array [
              Object {
                "color": "#231B22",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
                "marginBottom": 4,
                "marginTop": 4,
              },
            ]
          }
        >
          Feedback Message
        </Text>
      </View>
    </View>
  </View>,
]
`;
